### DISCUSSION TITLE
Missing required parameter for route

### DISCUSSION THREAD

**Discussion by yablochnyi** [filamentphp/filament#3222] [Help]
Hello, I can't pass the parameter. Where am I making a mistake?
![Снимок](https://user-images.githubusercontent.com/75489338/180709769-322bd1e7-42cf-4657-afcf-9c05e7909d09.JPG)
![Снимок2](https://user-images.githubusercontent.com/75489338/180709778-879a0d41-24a8-4b01-a070-4cba4cb9b48c.JPG)

**Comment by invaders-xx** [comment]
It's a bulkAction so you reveive a collection of records in your action fn(Collection $records) and not a record, that's why $record is null

**Reply by yablochnyi** [reply]
> It's a bulkAction so you reveive a collection of records in your action fn(Collection $records) and not a record, that's why $record is null
![Снимок](https://user-images.githubusercontent.com/75489338/180715463-fdcf18de-fd02-4537-b27b-8c25bd52adef.JPG)
I changed to a collection, but nothing happens now. Maybe I'm not launching the action correctly?
Route:
![Снимок2](https://user-images.githubusercontent.com/75489338/180715887-9a1c9b9e-c8ac-4721-8153-4e87738918f1.JPG)
Controller
![Снимок3](https://user-images.githubusercontent.com/75489338/180716069-58a062fe-781c-4031-a948-bac497aaec96.JPG)

**Reply by invaders-xx** [reply]
sure your controller is taking a product and not a collection ...

**Reply by yablochnyi** [reply]
> sure your controller is taking a product and not a collection ...

Yes you are right. After I wrote it, I found the problem)
